solutions/cf-1468D.mdx

```diff
@@ -0,0 +1,95 @@
+---
+id: cf-1468D
+source: CF
+title: Firecrackers
+author: Jesse Choe
+---
+
+[Official Editorial](https://assets.codeforces.com/rounds/1468/tutorials.pdf)
+
+We can initially attempt this problem by figuring out which firecrackers should explode. Obviously, we should attempt to explode the firecrackers with a minimal
+detonation time to increase the number of firecrackers exploded before eventually being caught. We can also observe that the firecrackers with a longer detonation 
+time should be dropped first.
+
+To find the maximum number of firecrackers before the hooligan gets caught by the guard, we can sort the detonation times in increasing order and figure out the
+maximum number of firecrackers to drop before getting caught by applying binary search.
+
+If we can determine the amount of time before getting caught, then the binary search will be trivial.
+
+There are two cases which determine the amount of time before getting caught:
+
+1. It can be proven that if $a < b$, then the amount of time before getting caught is precisely $b - 1$ seconds.
+2. It can be proven that if $a > b$, then the amount of time before getting caught is precisely $n - b$ seconds.
+
+**Note: these times are computed by assuming that both the hooligan and guard act optimally.**
+
+After computing the amount of time before getting caught, we can simulate whether a particular firecracker will explode in time.
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+<LanguageSection>
+
+<CPPSection>
+
+### C++ Implementation
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+using vl = vector<ll>;
+
+ll cooridor_size, num_of_firecrackers, hooligan_location, guard_location, maxFirecrackers, timeBeforeCaught;
+
+bool works(vl explodingTimes, ll firecrackers_exploded){
+	int current_time = 1;
+	for(int i = firecrackers_exploded - 1; i >= 0; i--){
+		if(current_time + explodingTimes[i] > timeBeforeCaught){
+			return false;
+		} else {
+			++current_time;
```

Comment:
i'd honestly prefer `current_time++` but that's just me

Reply:
yeah it doesn't really matter.